### PR TITLE
Implement content-sensitive canvas farbling

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/8661

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: chromium_src/crypto/hmac.cc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include "crypto/openssl_util.h"
+#include "crypto/secure_util.h"
+#include "crypto/symmetric_key.h"
+#include "third_party/boringssl/src/include/openssl/hmac.h"
+
+namespace crypto {
+
+// This function is identical to other Sign() function except it takes a raw
+// buffer and buffer length instead of a StringPiece.
+bool HMAC::Sign(unsigned char* buffer,
```
Comment (bridiver) [inline chromium_src/crypto/hmac.cc:23]: does this method have to be added to the actual hmac class? Can create some kind of utility method instead?

Comment (bridiver) [reply chromium_src/crypto/hmac.cc:23]: or alternatively can we convert the buffer to a stringpiece? I don't believe that anything is copied if you instantiate a new StringPiece from a char*

Comment (pilgrim-brave) [reply chromium_src/crypto/hmac.cc:23]: Well we could. It seemed cleaner to extend the class so we don't have to rewrite `Init` from scratch too.

Comment (bridiver) [reply chromium_src/crypto/hmac.cc:23]: `net::x509_util::CryptoBufferAsStringPiece` maybe?

**Comment by pilgrim-brave** [comment]
All CI failures are (finally) unrelated.

**Comment by bsclifton** [comment]
Added missing milestone (1.10) to this PR and issue (https://github.com/brave/brave-browser/issues/8661)

**Pull request merged**